ui/app/src/views/ViewDashboard.tsx

```diff
@@ -70,13 +71,15 @@ function ViewDashboard() {
   const isReadonly = useIsReadonly();
 
   // Collect the Project variables and setup external variables from it
-  // TODO: Once we'll implement global variables CRUD, we'll do the same with 'global' as source
+  const { data: globalVars, isLoading: isLoadingGlobalVars } = useGlobalVariableList();
   const { data: projectVars, isLoading: isLoadingProjectVars } = useVariableList(projectName);
   const externalVariableDefinitions: ExternalVariableDefinition[] | undefined = useMemo(() => {
     const result: ExternalVariableDefinition[] = [];
-    if (projectVars && projectVars.length > 0) result.push(buildExternalVariableDefinition('project', projectVars));
+    if (projectVars && projectVars.length > 0) result.push(buildProjectVariableDefinition(projectName, projectVars));
```

Comment:
For my own understanding, I’m not 100% following what all we’re doing with the response in these two functions and `buildExternalVariableDefinition`, are there any changes on the BE that could simplify this mapping in the future? I also noticed this comment which confused me on what is in the spec vs metadata:

*“The name of the variable resource can be missing from the spec but present in the metadata.”*

I’m also starting to wonder if the way we store separate kinds for GlobalVariable and ProjectVariable could be simplified somehow since we are mapping them to Variable anyways and using `source` but I guess we have to be explicit so that we know the hierarchy for which overrides the other?